### PR TITLE
Type definitions for happypack

### PR DESCRIPTION
**Pull request by akashishu777**

### CODE REVIEW

File: types/happypack/tslint.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
```
Comment (PranavSenthilnathan) [inline types/happypack/tslint.json]: Could you remove these rules?

File: types/happypack/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// Project: https://github.com/amireh/happypack
+// Definitions by: Akash Vishwakarma <https://github.com/akashishu777>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { Plugin } from 'webpack';
+
+declare class HappyPack extends Plugin {
+
+    constructor(options?: HappyPack.PluginOptions);
```
Comment (PranavSenthilnathan) [inline types/happypack/index.d.ts]: The docs and source seem to indicate that this is a should be a required parameter.

File: types/happypack/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+declare namespace HappyPack {
+    interface PluginOptions {
+        id?: string;
+        threads?: number;
+        loaders?: any;
```
Comment (PranavSenthilnathan) [inline types/happypack/index.d.ts]: `loaders` seems like a mandatory property.

File: types/happypack/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    }
+}
+
+export = HappyPack;
```
Comment (PranavSenthilnathan) [inline types/happypack/index.d.ts]: Some tests to show how this library should be used in TS would be great.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/happypack/tslint.json (hunk 1 above)
Comment (akashishu777) [reply types/happypack/tslint.json]: Removed the rules.

File: types/happypack/index.d.ts (hunk 2 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: options are now required.

File: types/happypack/index.d.ts (hunk 3 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: loaders is now mandetory

File: types/happypack/index.d.ts (hunk 4 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: Added test.

File: types/happypack/index.d.ts (hunk 2 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: I have updated the code now options are required.

File: types/happypack/index.d.ts (hunk 3 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: made it mandatory property.

File: types/happypack/index.d.ts (hunk 4 above)
Comment (akashishu777) [reply types/happypack/index.d.ts]: Added test for testing.

**Review by PranavSenthilnathan — APPROVED** [review]

**Pull request merged**